src/main/scala/com/nec/spark/SparkCycloneExecutorPlugin.scala

```diff
@@ -20,180 +20,83 @@
+    require(pluginContext != null, s"${classOf[PluginContext].getSimpleName} has not been set yet!")
+    VeProcess.createFromContext(pluginContext)
+  }
+
+  implicit def source: VeColVectorSource = {
+    veProcess.source
+  }
+
   @transient implicit lazy val veMetrics: ProcessExecutorMetrics = {
     require(pluginContext != null, s"${classOf[PluginContext].getSimpleName} is not yet set!")
     new ProcessExecutorMetrics(AllocationTracker.simple, pluginContext.metricRegistry)
   }
 
-  var CleanUpCache: Boolean = true
+  @transient implicit lazy val vectorEngine: VectorEngine = {
+    require(pluginContext != null, s"${classOf[PluginContext].getSimpleName} is not yet set!")
+    new VectorEngineImpl(veProcess, pluginContext.metricRegistry)
+  }
 
   @transient val batchesCache = new VeColBatchesCache
 }
 
 class SparkCycloneExecutorPlugin extends ExecutorPlugin with Logging with LazyLogging {
-  import com.nec.spark.SparkCycloneExecutorPlugin._
-
-  override def init(ctx: PluginContext, extraConf: util.Map[String, String]): Unit = {
-    val resources = ctx.resources()
-
-    logger.info(s"Executor has the following resources available => ${resources}")
-    val selectedVeNodeId = if (!resources.containsKey("ve")) {
-      val nodeId = Try(System.getenv("VE_NODE_NUMBER").toInt).getOrElse(DefaultVeNodeId)
-      logger.info(
-        s"Do not have a VE resource available. Will use '${nodeId}' as the main resource."
-      )
-      nodeId
-    } else {
-      val veResources = resources.get("ve")
-      val resourceCount = veResources.addresses.length
-      SparkCycloneExecutorPlugin.NodeCount = resourceCount
-      val executorNumber = Try(ctx.executorID().toInt - 1).getOrElse(0) // Executor IDs start at 1.
-      val veMultiple = executorNumber / 8
-      if (veMultiple > resourceCount) {
-        logWarning("Not enough VE resources allocated for number of executors specified.")
-      }
-      veResources.addresses(veMultiple % resourceCount).toInt
-    }
+  override def init(context: PluginContext, conf: JMap[String, String]): Unit = {
+    logger.info(s"Initializing ${getClass.getSimpleName}...")
+
+    // Set the plugin context
+    SparkCycloneExecutorPlugin.pluginContext = context
 
-    logger.info(s"Using VE node = ${selectedVeNodeId}")
+    // Update the plugin params
+    SparkCycloneExecutorPlugin.params ++= conf.asScala
 
-    var currentVeNodeId = selectedVeNodeId
-    while (_veo_proc == null && currentVeNodeId < 8) {
-      _veo_proc = veo.veo_proc_create(currentVeNodeId)
-      if (_veo_proc == null) {
-        logWarning(s"Proc could not be allocated for node ${currentVeNodeId}, trying next.")
-        currentVeNodeId += 1
-      }
-      _veo_thr_ctxt = veo.veo_context_open(_veo_proc)
+    // Update the node count as needed
+    SparkCycloneExecutorPlugin.NodeCount = {
+      val resources = context.resources
+      if (resources.containsKey("ve")) resources.get("ve").addresses.length else 1
```

Comment:
In the new system where does the VeNodeId logic live?  Does it now only rely on veo_proc_create(-1)?

Reply:
This lives inside `VeProcess.scala` - there are different overloads for `VeProcess.create` defined there.